apps/backend/src/statusActionEngine/emailActionHandler.ts

```diff
@@ -21,203 +22,379 @@ import {
+  ) => Promise<void>,
+  statusActionsLogId?: number | null,
+  loggedInUserId?: number | null
+) => {
   const mailService = container.resolve<MailService>(Tokens.MailService);
+  const successfulMessage = !!statusActionsLogId
+    ? 'Email successfully sent on status action replay'
+    : 'Email successfully sent';
+  const failMessage = !!statusActionsLogId
+    ? 'Email(s) could not be sent on status action replay'
+    : 'Email(s) could not be sent';
+
+  if (!recipientsWithData.length) {
+    logger.logInfo('Could not send email(s) because there are no recipients.', {
+      recipientsWithData,
+    });
 
-  await Promise.all(
-    recipientsWithData.map(async (recipientWithData) => {
-      return mailService
-        .sendMail({
-          content: {
-            template_id: recipientWithData.template,
-          },
-          substitution_data: {
-            proposals: recipientWithData.proposals,
-            pi: recipientWithData.pi,
-            coProposers: recipientWithData.coProposers,
-            instruments: recipientWithData.instruments,
-            // The firstName, lastName, preferredName of the main recipient
-            firstName: recipientWithData.firstName,
-            lastName: recipientWithData.lastName,
-            preferredName: recipientWithData.preferredName,
-          },
-          recipients: [{ address: recipientWithData.email }],
-        })
-        .then(async (res) => {
+    return;
+  }
+  try {
+    const mailServiceResponse = await Promise.all(
+      recipientsWithData.map(async (recipientWithData) => {
+        try {
+          const res = await mailService.sendMail({
+            content: {
+              template_id: recipientWithData.template,
+            },
+            substitution_data: {
+              proposals: recipientWithData.proposals,
+              pi: recipientWithData.pi,
+              coProposers: recipientWithData.coProposers,
+              instruments: recipientWithData.instruments,
+              firstName: recipientWithData.firstName,
+              lastName: recipientWithData.lastName,
+              preferredName: recipientWithData.preferredName,
+            },
+            recipients: [{ address: recipientWithData.email }],
+          });
           logger.logInfo('Email sent:', {
             result: res,
           });
 
-          const messageDescription = `Email successfully sent to: ${recipientWithData.email}`;
           await publishMessageToTheEventBus(
             recipientWithData.proposals,
-            messageDescription
+            `${successfulMessage} to: ${recipientWithData.email} recipient: ${recipientWithData.id}`,
+            undefined,
+            loggedInUserId || undefined
           );
-        })
-        .catch((err) => {
+
+          return res;
+        } catch (err) {
           logger.logError('Could not send email', {
             error: err,
           });
-        });
-    })
-  );
+
+          await publishMessageToTheEventBus(
```

Comment:
Could you also add this in the PR description, so that it gets logged. 

Publishing message in the event of email failure.